src/aws-cur/cost-report.ts

```diff
@@ -0,0 +1,102 @@
+import { Stack, aws_cur, aws_iam, aws_s3 } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+
+export class ReportGranularity {
+
+  public static readonly HOURLY: ReportGranularity = new ReportGranularity('HOURLY');
+  public static readonly DAILY: ReportGranularity = new ReportGranularity('DAILY');
+  public static readonly MONTHLY: ReportGranularity = new ReportGranularity('MONTHLY');
+
+  private constructor(public readonly value: string) { }
+}
+
+export class CurFormat {
+
+  public static readonly TEXT: CurFormat = new CurFormat('GZIP', 'textORCsv');
+  public static readonly PARQUET: CurFormat = new CurFormat('Parquet', 'Parquet');
+
+  private constructor(public readonly compression: string, public readonly format: string) { }
```

Comment:
This talk is similar to @mbonig 's [comment](https://github.com/open-constructs/aws-cdk-library/pull/15/files#r1569627669)...

When exporting this class, it is conceivable that users may extend it. I think it would be better to make it `protected` not `private` to take this into account. For example, some users may want to choose `ZIP` for `compression` instead of `GZIP`.

```suggestion
  protected constructor(public readonly compression: string, public readonly format: string) { }
```

Reply:
What about making it public then or creating a factory method?

Reply:
Oops. I see the former with `public` was in the guidelines as well. Either way, I like the factory method.

Reply:
I implemented it this way.